datastore/datastore.go

```diff
@@ -185,11 +186,14 @@ func ParseKey(key string) ([]string, error) {
 }
 
 // newClient used to connect to KV Store
-func newClient(scope string, kv string, addr string, config *store.Config, cached bool) (DataStore, error) {
+func newClient(scope string, kv string, addr string, config *store.Config, cached bool, sequential bool) (DataStore, error) {
 
 	if cached && scope != LocalScope {
 		return nil, fmt.Errorf("caching supported only for scope %s", LocalScope)
 	}
+	if scope == LocalScope && !sequential {
```

Comment:
If this function is aware that local scope => sequential, then let's not have the caller set the sequential flag, rather this function itself can do it by checking the scope. Whether to set the sequential flag or not  is anyway an internal decision in this package, given `newClient()` is not exported, then might as well this function do it itself. And no consistency check needed.